volatility3/framework/plugins/linux/proc.py

```diff
@@ -35,16 +40,138 @@ def get_requirements(cls):
+        task: interfaces.objects.ObjectInterface,
+        vma: interfaces.objects.ObjectInterface,
+        open_method: Type[interfaces.plugins.FileHandlerInterface],
+        maxsize: int = MAXSIZE_DEFAULT,
+    ) -> Optional[interfaces.plugins.FileHandlerInterface]:
+        """Extracts the complete data for VMA as a FileInterface.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            task: an task_struct instance
+            vma: The suspected VMA to extract (ObjectInterface)
+            open_method: class to provide context manager for opening the file
+            maxsize: Max size of VMA section (default MAXSIZE_DEFAULT)
+
+        Returns:
+            An open FileInterface object containing the complete data for the task or None in the case of failure
+        """
+        try:
+            vm_start = vma.vm_start
+            vm_end = vma.vm_end
+        except AttributeError:
+            vollog.debug("Unable to find the vm_start and vm_end")
+            return None
+        
+        vm_size = vm_end - vm_start
+        if 0 < maxsize < vm_size:
+            vollog.debug(
+                f"Skip virtual memory dump {vm_start:#x}-{vm_end:#x} due to maxsize limit"
+            )
+            return None
+
+        pid = "Unknown"
+        try:
+            pid = task.tgid
+            proc_layer_name = task.add_process_layer()
+        except exceptions.InvalidAddressException as excp:
+            vollog.debug(
+                "Process {}: invalid address {} in layer {}".format(
+                    pid, excp.invalid_address, excp.layer_name
+                )
+            )
+            return None
+
+        proc_layer = context.layers[proc_layer_name]
+        file_name = f"pid.{pid}.vma.{vm_start:#x}-{vm_end:#x}.dmp"
+        try:
+            file_handle = open_method(file_name)
+            chunk_size = 1024 * 1024 * 10
+            offset = vm_start
+            while offset < vm_start + vm_size:
+                to_read = min(chunk_size, vm_start + vm_size - offset)
+                data = proc_layer.read(offset, to_read, pad=True)
+                if not data:
+                    break
+                file_handle.write(data)
+                offset += to_read
+
+        except Exception as excp:
+            vollog.debug(f"Unable to dump virtual memory {file_name}: {excp}")
+            return None
+
+        return file_handle
+
     def _generator(self, tasks):
+        # build filter for addresses if required
+        address_list = self.config.get("address", [])
+        if address_list == []:
+            # do not filter as no address_list was supplied
+            filter_func = lambda _: False 
+        else:
+            # filter for any vm_start that matches the supplied address config
+            def filter_function(x: interfaces.objects.ObjectInterface) -> bool:
+                return x.vm_start not in address_list
```

Comment:
I think this would also need changing to flip the filter function.

Reply:
This would also be where the change to allow segment-internal-addresses to act as filters would be.  A little more complex, but should be doable.